[2/N] Add option context

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11865
* #11391
* __->__ #11390
* #11389



For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

cccclai committed:
[2/N] Add update context

For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add update context"


For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add update context"


For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add update context"


For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add update context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add update context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

kimishpatel reviewed (commented):
I am not sure on the use of this. I understand you said this is introduced for BC, but can you elaborate on the BC aspect of it? Is it to avoid makign changes to client call sites? If this struct is populated then I imagine it will require client callsites to change anyway, if they want to make use of it. So maybe a bit more on this in the summary would be good.

If you do want to keep it then maybe get_option and set_option should provide both overloads? One with backendoptioncontext and one without..

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]

cccclai committed:
Update on "[2/N] Add option context"




For future needs without breacking API BC, in case we need to pass more information to the update API

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212/)

Differential Revision: [D75919212](https://our.internmc.facebook.com/intern/diff/D75919212)

[ghstack-poisoned]
